frontend/src/metabase/visualizations/visualizations/CartesianChart/CartesianChart.tsx

```diff
@@ -115,13 +171,52 @@ export function CartesianChart({
     ],
     [
       chartModel,
+      onHoverChange,
       onVisualizationClick,
       openQuestion,
       settings,
       visualizationIsClickable,
     ],
   );
 
+  useEffect(() => {
```

Comment:
It would be better if we can make this a custom hook and name it, e.g. `useChartHighlight`, right now it's not easy to tell what this hook does and what it's purpose is without reading the whole thing. It would also help better organize the code by separating responsibilities into different files, a file structure tree is easier to parse and find what you are looking for in rather than one very long file that has many different responsibilities

Reply:
Will extract on top of the stack since the code is slightly changing